x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrLicenseIT.java

```diff
@@ -84,15 +85,39 @@ public void onResponse(final CcrStatsAction.TasksResponse tasksResponse) {
 
             @Override
             public void onFailure(final Exception e) {
-                assertIncompatibleLicense(e);
+                assertNonCompliantLicense(e);
                 latch.countDown();
             }
         });
 
         latch.await();
     }
 
-    private void assertIncompatibleLicense(final Exception e) {
+    public void testThatPutAutoFollowPatternsIsUnavailableWithNonCompliantLicense() throws InterruptedException {
+        final CountDownLatch latch = new CountDownLatch(1);
+        final PutAutoFollowPatternAction.Request request = new PutAutoFollowPatternAction.Request();
+        request.setLeaderClusterAlias("leader");
+        request.setLeaderIndexPatterns(Collections.singletonList("*"));
+        client().execute(
+                PutAutoFollowPatternAction.INSTANCE,
+                request,
+                new ActionListener<AcknowledgedResponse>() {
+                    @Override
+                    public void onResponse(final AcknowledgedResponse response) {
+                        fail();
```

Comment:
do we need to invoke `latch.countDown()` here? Otherwise in case of failure this test would hang for a while before failing?

Reply:
I pushed d3d6412f81b793ce80fd56f74a0cf7ae0cba9da9.